app/importers/file_importers/student_section_assignments_importer.rb

```diff
@@ -38,13 +38,19 @@ def filter
     SchoolFilter.new(@school_scope)
   end
 
-  def delete_rows
-    #Delete all stale rows no longer included in the import
-    #For the schools imported during this run of the importer
+  def rows_not_imported
     StudentSectionAssignment.joins(:section => {:course => :school})
                             .where.not(id: @imported_assignments)
-                            .where(:schools => {:local_id => @school_scope}).delete_all
+  end
+
+  def rows_to_delete
+    return rows_not_imported if school_scope.nil?
 
+    rows_not_imported.where(:schools => {:local_id => @school_scope})
+  end
+
+  def delete_rows
+    rows_to_delete.delete_all
```

Comment:
@alexsoble I don't really understand why this change is related, but this seems reasonable.  This might be worth explaining at the top of the importer class - this behavior is slightly different than the others and not obvious until you read these particular methods how changing school scope will impact what rows are deleted (not just which ones are imported).  Also probably worth spot checking after this ships.

Reply:
@kevinrobinson Yep, it's really not related. This change was motivated by the StudentSectionAssignmentsImporter spec. Changing the schools factory broke the spec. I dove in and found the spec was breaking because of its `@school_scope`. This change to the feature-side code was the quickest way I could get it to pass and seemed like a reasonable change . . . I guess that's a bit hacky though huh? 😄 

I'll try to reproduce the spec failure up on Travis and maybe we'll find a better way to address it.

Reply:
This Travis failure shows the exact same spec error, except in `educator_section_assignments_importer_spec.rb` instead of `student_section_assignments_importer_spec.rb`: 

https://travis-ci.org/studentinsights/studentinsights/builds/338115130?utm_source=github_status&utm_medium=notification

Reply:
Here's the error: 

```
$ bundle exec rspec spec
..........................................................................................................................................................................................................................F.............................................................................................................................................................................................................................................................
Failures:
  1) EducatorSectionAssignmentsImporter#delete_rows happy path deletes all student section assignments except the recently imported one
     Failure/Error: expect(EducatorSectionAssignment.count).to eq(1)
     
       expected: 1
            got: 21
     
       (compared using ==)
     # ./spec/importers/file_importers/educator_section_assignments_importer_spec.rb:133:in `block (4 levels) in <top (required)>'
```

Reply:
Ooooh, I dove in deeper and have a hunch that this has to do with the `SchoolFilter` class. The `SchoolFilter` uses school local IDs to do filtering, but on `master`, all schools created by our factories have a `local_id` of `nil`, which isn't a very realistic data setup. My current hypothesis is that giving those factory schools real local IDs changes the way that school filtering interacts with the importer in this spec setup.

Reply:
@kevinrobinson I was wrong about this actually! Turns out `SchoolFilter` is never called in these specs, because the specs don't test `#import`, they skip it and go straight to testing `#import_row` methods, which assume the rows have already been filtered. 

I found that the issue had to do with the test instances of `EducatorSectionAssignmentsImporter` and `StudentSectionAssignmentsImporter` we were testing: these had `{school_scope: nil}`, which worked when every school created by FactoryGirl had a local_id of `nil`, but started failing once the FactoryGirl schools were assigned local_ids.